docs/usage.rst

```diff
@@ -359,6 +359,15 @@ Use it to extract just the first matching string::
     >>> selector.xpath('//a[contains(@href, "image")]/text()').re_first(r'Name:\s*(.*)')
     'My image 1 '
 
+You can also use compiled regular expressions with both methods::
+
+    >>> import re
+    >>> regex = re.compile(r'Name:\s*(.*)')
+    >>> selector.xpath('//a[contains(@href, "image")]/text()').re_first(regex)
+    'My image 1 '
+
+As well as adding regex flags with the ``flags`` argument.
```

Comment:
I think this needs an example and/or a link to the Python doc about the flags.

Reply:
I would also be OK with just appending ``` (see :mod:`re`) ``` to the first sentence of this entire section, right after the first mention of “regular expressions”, since at that point already users may wonder about regular expressions.